src/metaopt/io/database/mongodb.py

```diff
@@ -11,17 +11,9 @@
 except ImportError:
     from io import StringIO
 
-try:
-    # Python 3.x
-    from urllib.parse import quote_plus
-except ImportError:
-    # Python 2.x
-    from urllib import quote_plus
-
 class MongoDB(AbstractDB):
-    def __init__(self, dbhost='localhost', dbname='metaopt',
-                dbport=27017, username=None, password=None):
-        super(MongoDB, self).__init__(dbhost, dbname, dbport, username, password)
+    def __init__(self, uri):
```

Comment:
It seems my comment on slack wasn't clear. You did the exact opposite of what I proposed. :laughing: 

I believe the simplest functional interface for now would be `_init__(uri, with, all, the, sugar, already, there, being, optional)`, but inside `__init__` only `self._uri = uri` would be set. Then, the uri getter would be as you wrote. However, there would be no setter for now as all the sugar already there are not set in the database yet. 

I know it's tempting to already implement all of it. It's what, just 2 or 3 more functions and a couple of attributes after all? But then we need to test all use cases and make sure passing uri works fine just as much as passing attributes, and think about what to do if user pass uri + other arguments. How do we handle the setter for uri? How do we handle the setter for the other arguments? It's a lot more work than it looks like, enough to make us miss the Friday's milestone for the DB. Furthermore, we would have to maintain this code while implementing the rest of the minimum viable product, hence spending even more time maintaining extra features.

Reply:
I don't get it really. If you mean we use the function parser, we need to pass a `uri` into it, before it was like this, it was `__init__(self, dbhost, dbname, dbport, username, password))` which you said we shouldn't do this and we should just use the parser instead. If that get's parsed again inside the `MongoClient` we should just pass in the `uri`, which it would look like just that. 

I don't know how this is incorrectly. You don't need to pass through init both `uri` and all of the `dbhost`, `dbname`, `dbpassword`, and `dbport` all over again, it's inefficient.

Reply:
I see why you are confused, I'm sorry I didn't detail enough my thoughts. 

I believe we should eventually support both 

```python
db = Database(uri)
```

and 

```python
db = Database(dbhost, dbname, dbport, username, password, **options)
```

Just like pymongo's MongoClient does. But for now only one of the two would be enough for the MVP. I believe `Database(uri)` would be the simplest one to implement first. You can keep the arguments in the init as they are optional, but they wouldn't be used for now.

We could use the same mechanism than MongoClient:
> class pymongo.mongo_client.MongoClient(host='localhost', port=27017, ...)
> ...
> The host parameter can be a full mongodb URI, in addition to a simple hostname